include/seqan3/core/platform.hpp

```diff
@@ -90,7 +90,39 @@
 #   error The range-v3 library was not included correctly. Forgot to add -I ${INSTALLDIR}/include to your CXXFLAGS?
 #endif
 
-// SDSL
+// SDSL [required]
+// TODO (doesn't have a version.hpp, yet)
+
+// Cereal [optional]
+/*!\def SEQAN_WITH_CEREAL
+ * \brief Whether CEREAL support is available or not.
+ * \ingroup core
+ */
+#ifndef SEQAN_WITH_CEREAL
+#   if __has_include(<cereal/cereal.hpp>)
+#       define SEQAN_WITH_CEREAL 1
+#   else
+#       define SEQAN_WITH_CEREAL 0
+#   endif
+#endif
+
+#if !SEQAN_WITH_CEREAL
+    /*!\cond DEV
+     * \name Cereal function macros
+     * \ingroup core
+     * \brief These can be changed by apps so we used the macros instead of the values internally.
```

Comment:
This is part of CEREAL's design. It allows you to incorporate cereal in your library and avoid naming conflicts in case any of the mentioned function names is already used for something else. We just make sure that if there ever is someone who uses this feature, it will work with seqan, too. 
Our documentation system automatically replaces these macros with simple function names, though, so it's no problem for our users (they don't use these functions directly anyway).